uportal-war/src/main/java/org/jasig/portal/rest/PeopleRESTControllerV43.java

```diff
@@ -0,0 +1,124 @@
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.ModelAndView;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Controller
+@RequestMapping("/v4-3/people/")
+public class PeopleRESTControllerV43 {
+
+    private IPersonManager personManager;
+    private ObjectMapper jsonMapper;
+
+    @Autowired
+    public void setJsonMapper(ObjectMapper jsonMapper) {
+        this.jsonMapper = jsonMapper;
+    }
+
+    @Autowired
+    public void setPersonManager(IPersonManager personManager) {
+        this.personManager = personManager;
+    }
+
+    private PersonLookupHelperImpl lookupHelper;
+
+    @Autowired(required = true)
+    public void setPersonLookupHelper(PersonLookupHelperImpl lookupHelper) {
+        this.lookupHelper = lookupHelper;
+    }
+
+    @RequestMapping(value="/", method = RequestMethod.GET)
+    public void searchPeople(@RequestParam Map<String,Object> query,
+            HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        final IPerson user = personManager.getPerson((HttpServletRequest) request);
+        if (user == null) {
+            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+            return;
+        }
+
+        List<IPersonAttributes> people = lookupHelper.searchForPeople(user, query);
+
+        if ( people == null ) {
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            return;
+        }
+
+        //Unwrap attributes
+        List<Object> results = new ArrayList<Object>();
+        for(IPersonAttributes p : people) {
+            results.add(p.getAttributes());
+        }
+
+        response.setHeader("Content-Type", "application/json");
+        jsonMapper.writeValue(response.getOutputStream(), results);
+    }
+
+    @RequestMapping(value="/{username}", method = RequestMethod.GET)
+    public ModelAndView getPerson(@PathVariable String username,
+            HttpServletRequest request, HttpServletResponse response) {
+
+        final IPerson searcher = personManager.getPerson(request);
+        if (searcher == null) {
+            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+            return null;
+        }
+
+        final IPersonAttributes person = lookupHelper.findPerson(searcher, username);
+        if (person == null) {
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            return null;
+        }
+
+        return new ModelAndView("json", person.getAttributes());
+    }
+
+    @RequestMapping(value = "/me", method = RequestMethod.GET)
```

Comment:
After some contemplation, I suggest we start dropping the extensions, like ".json". There are a number of formats coming out which may challenge JSON in the future. I prefer our APIs  with well known URIs to respond intelligently rather than muddy them with variations for the same endpoint.